### DISCUSSION TITLE
Add button on top of Table

### DISCUSSION THREAD

**Discussion by croustibat** [filamentphp/filament#1807] [Help] [answered]
Hello,

I can't figure out how to add a button on top of the Table in a RelationManager class : 

<img width="1118" alt="Screenshot 2022-03-06 at 16 25 15" src="https://user-images.githubusercontent.com/1169456/156929792-7c793c35-2457-4f00-bd72-7bf3b58e0f80.png">

Thanks for your help,

**Comment by danharrin** [comment, marked as answer]
Something like this?

```php
protected function getTableHeaderActions(): array
{
    return array_merge(parent::getTableHeaderActions(), [
        ButtonAction::make('name')->action(fn () => // do something)
    ]);
}
```

**Reply by croustibat** [reply]
That's it ! Thank you so much 🙏

**Reply by sh4msi** [reply]
Thankful
Why are these not in the documentation?

**Reply by danharrin** [reply]
It's actually much easier now, just add the Action to the `->headerActions([...])` of the table. No overriding methods
